### PR TITLE
[EPO-7920] News & Events callout

### PR DESCRIPTION
**Pull request by blnkt**
New callouts tailored to News and Events entry types.  Was able to reuse the same component and styles for both (combo!).  Should not change the style or functionality of the other callout types. See Jose's designs attached.  

[20230405 New Callouts - Desktop.pdf](https://github.com/lsst-epo/rubin-obs-client/files/11445573/20230405.New.Callouts.-.Desktop.pdf)

[20230405 Callouts —320px.pdf](https://github.com/lsst-epo/rubin-obs-client/files/11445571/20230405.Callouts.320px.pdf)

[20230405 New Callouts - 720 px.pdf](https://github.com/lsst-epo/rubin-obs-client/files/11445572/20230405.New.Callouts.-.720.px.pdf)

To test locally you need `rubin-oba-api` on branch `EPO-7919` (https://github.com/lsst-epo/rubin-obs-api/pull/147).  First, create new callouts with Entry Type "Callout - News" and "Callout - Events", selecting existing news and event entries for each: observe you only have options to add and remove the entry, and to change the color appearance.  Next, add your callouts to a page: observe that they visually match the provided designs.

### CODE REVIEW

File: components/content-blocks/Callout/CalloutEntry/index.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      lang
+    )}`;
+  }
+
+  if (eventEnd) {
+    return makeDateString(eventEnd, lang);
+  }
+};
+
+export default function CalloutEntry({ callout }) {
```
Comment (blnkt) [inline components/content-blocks/Callout/CalloutEntry/index.js:33]: Found it was prudent for both Event and News callouts to be rendered by the same component, "Entry" felt then like the relevant term to include in the component name.

File: components/content-blocks/Callout/CalloutEntry/styles.js [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+  respond,
+  tokens,
+} from "@/styles/globalStyles";
+import { Section as CalloutSection } from "../styles";
+
+export const Section = styled(CalloutSection)`
+  position: relative;
+`;
+
+export const Inner = styled(MixedLink)`
```
Comment (blnkt) [inline components/content-blocks/Callout/CalloutEntry/styles.js:16]: This `Section` is identical to the `Section` in `/content-blocks/Callout`, except for the positioning.  This felt like the styled componenty way to accomplish this, but maybe you have a better pattern?  `Inner` was similar to the `Inner` in `/content-blocks/Callout`, but then departed enough from it that I felt like it made sense to have it be its own thing (i.e. not extended from the other `Inner`).

File: components/layout/SharePopup/styles.js [hunk 3]
```diff
@@ -1,8 +1,20 @@
...
+import { Popover } from "@headlessui/react";
 
 export const PanelInner = styled.div`
   padding: 0.5em;
   background-color: var(--white);
   border-radius: 0.5em;
   box-shadow: 5px 15px 35px 8px rgba(0, 0, 0, 0.13);
 `;
+
+export const Button = styled(Popover.Button)`
```
Comment (blnkt) [inline components/layout/SharePopup/styles.js:11]: A small unrelated fix to the lil share menu: it didn't have any visible hover/focus/open styling.  Decided a simple outline felt the most appropriate and flexible to how many background colors this button gets used over.

File: components/content-blocks/Callout/CalloutEntry/styles.js [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+  }
+
+  ${respond(
+    `
+    grid-template-areas:
+      "image image"
+      "title title"
+      "subtitle subtitle"
+      "text text"
+      "footer footer";
```
Comment (blnkt) [inline components/content-blocks/Callout/CalloutEntry/styles.js:93]: This may be the first CSS grid I ever made from scratch.  Middling confidence that I did it all correctly, though it seems to be working as expected on the frontend.

File: components/content-blocks/Callout/CalloutEntry/index.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+import {
+  Section,
+  Inner,
+  SharePopup,
+  ImageWrapper,
+  ImageSticker,
+} from "./styles";
+import { Image } from "@rubin-epo/epo-react-lib";
+
+const getDateString = (newsDate, eventStart, eventEnd, lang) => {
```
Comment (blnkt) [inline components/content-blocks/Callout/CalloutEntry/index.js:16]: lil date making helper to negotiate news post date vs event date vs event date with start & end dates

File: components/content-blocks/Callout/CalloutEntry/index.js [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    image,
+    entryType,
+  } = entry[0];
+  const { title: type, slug: typeSlug } = entryType[0];
+  const titleId = `${typeSlug}-${id}`;
+  const calloutDateString = getDateString(date, startDate, endDate, lang);
+
+  return (
+    <Section $bgColor={backgroundColor} $width="full" $overlay={false}>
+      <Inner href={uri} aria-labelledby={titleId}>
```
Comment (blnkt) [inline components/content-blocks/Callout/CalloutEntry/index.js:55]: Using these explicitly in this Callout variant because the Callout `Wrapper` used in the other callouts did not easily allow for customizing the `Section` and `Inner` to the extent needed, and also needed the `SharePopup` to be inside the `Section` but outside the `Inner` to get the clickability and positioning we needed (conversely, `Wrapper` all children get wrapped in `Section` and `Inner`).

File: components/content-blocks/Callout/CalloutEntry/styles.js [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+export const ImageWrapper = styled.div`
+  position: relative;
+`;
+
+export const ImageSticker = styled.div`
+  position: absolute;
+  top: 0;
+  left: 0;
+  padding: 16px 24px;
+  color: ${tokens.white};
```
Comment (alexgoff) [inline components/content-blocks/Callout/CalloutEntry/styles.js:183]: everything in tokens can just be written as a CSS var

File: components/content-blocks/Callout/CalloutEntry/styles.js [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+  )}
+`;
+
+export const SharePopup = styled(SharePopupComponent)`
+  position: absolute;
+  inset-inline-end: 30px;
+  inset-block-end: 30px;
+  color: var(--neutral40);
+
+  @media (max-width: ${tokens.BREAK_TABLET}) {
```
Comment (alexgoff) [inline components/content-blocks/Callout/CalloutEntry/styles.js:203]: actually this one can't be a CSS var because media queries aren't attached to HTML elements

File: components/content-blocks/Callout/CalloutEntry/styles.js (hunk 4 above)
Comment (alexgoff) [reply components/content-blocks/Callout/CalloutEntry/styles.js:93]: looks good!

File: components/content-blocks/Callout/CalloutEntry/styles.js [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+  min-height: 360px;
+  height: 100%;
+  padding-top: ${fluidScale("50px", "30px")};
+  padding-bottom: ${fluidScale("50px", "40px")};
+  align-content: start;
+  text-decoration: none;
+  transition: color 0.2s, background-color 0.2s;
+
+  // set the grid areas for various bits
+  .image {
```
Comment (alexgoff) [inline components/content-blocks/Callout/CalloutEntry/styles.js:39]: you can do this without setting additional classnames, since `ImageWrapper` is being defined in this styles module you can replace

```
.images { ... }
```

with 

```
${ImageWrapper} {...}
```

and styled-components will extract the generated classname for ImageWrapper and use that

**Review by alexgoff — CHANGES_REQUESTED** [review]
I left some notes on styled-components cleanup, the functionality is good, I was able to step through and create the new callout types just fine!
